From f1444e3101c9b3b4263d8fa82cfbadb0a9b4db28 Mon Sep 17 00:00:00 2001
From: Jan Stancek <jstancek@redhat.com>
Date: Tue, 7 Jul 2020 16:07:14 +0200
Subject: [PATCH 3/3] runltp: let each test acquire/release device

Problem with runltp creating and using single loop device for all tests
is that if one test fails/gets stuck or fails to umount loop device
for any reason, then all subsequent tests also fail, because
they try to use mounted device.

Don't force same loop device on all tests (unless it is chosen via
env. variable DEVICE) and let LTP library try to acquire first
usable loop device for each test.

Fixes #703
Signed-off-by: Jan Stancek <jstancek@redhat.com>
Acked-by: Cyril Hrubis <chrubis@suse.cz>

Upstream-Status: Backport

Reference to upstream patch:
https://github.com/linux-test-project/ltp/commit/fec810aa86296cf230b672e290ec41c70fff4dd8

Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
 runltp | 38 +-------------------------------------
 1 file changed, 1 insertion(+), 37 deletions(-)

diff --git a/runltp b/runltp
index daa9be5a3..d615390e4 100755
--- a/runltp
+++ b/runltp
@@ -939,45 +939,9 @@ EOF
     exit $VALUE
 }
 
-create_block()
-{
-    #create a block device
-    dd if=/dev/zero of=${TMP}/test.img bs=1024 count=262144 >/dev/null 2>&1
-    if [ $? -ne 0 ]; then
-        echo "Failed to create loopback device image, please check disk space and re-run"
-        return 1
-    else
-        ##search for an unused loop dev
-        LOOP_DEV=$(losetup -f)
-        if [ $? -ne 0 ]; then
-            echo "no unused loop device is found"
-            return 1
-        else
-            ##attach the created file to loop dev.
-            losetup $LOOP_DEV ${TMP}/test.img
-            if [ $? -ne 0 ]; then
-                echo "losetup failed to create block device"
-                return 1
-            fi
-            DEVICE=$LOOP_DEV
-            return 0
-        fi
-    fi
-}
-
 set_block_device()
 {
-    if [ -z "$DEVICE" ]; then
-        create_block
-        if [ $? -ne 0 ]; then
-            echo "no block device was specified on commandline."
-            echo "Block device could not be created using loopback device"
-            echo "Tests which require block device are disabled."
-            echo "You can specify it with option -b"
-	else
-            export LTP_DEV=$DEVICE
-        fi
-    else
+    if [ -n "$DEVICE" ]; then
         export LTP_DEV=$DEVICE
     fi
 }
-- 
2.25.1

